drivers/SmartThings/zigbee-button/src/dimming-remote/init.lua

```diff
@@ -37,14 +38,44 @@ local function can_handle_zigbee_dimming_remote(opts, driver, device, ...)
 end
 
 local function button_pushed_handler(button_number)
-  return function(self, device, value, zb_rx)
-    button_utils.init_button_press(device, button_number)
+  return function(driver, device, value, zb_rx)
+    local event = capabilities.button.button.pushed({ state_change = true })
+    local comp = device.profile.components[string.format("button%d", button_number)]
+
+    device:emit_event(event)
+    device:emit_component_event(comp, event)
+  end
+end
+
+local function button_hold_handler(button_number)
+  return function(driver, device, value, zb_rx)
+    device:set_field(string.format(BUTTON_X_HOLD, button_number), os.time())
+  end
+end
+
+local function get_button_hold_time(device, button_number)
+  local button_x_hold_time = device:get_field(string.format(BUTTON_X_HOLD, button_number))
+  local current_time = os.time()
+
+  return button_x_hold_time and current_time - button_x_hold_time or nil
+end
+
+local function send_held_event_if_applicable(device, button_number)
+  local event = capabilities.button.button.held({ state_change = true })
+  local comp = device.profile.components[string.format("button%d", button_number)]
+  local button_hold_time = get_button_hold_time(device, button_number)
+
+  device:set_field(string.format(BUTTON_X_HOLD, button_number), nil)
+
+  if button_hold_time and button_hold_time < TIMEOUT_THRESHOLD then
+    device:emit_event(event)
+    device:emit_component_event(comp, event)
```

Comment:
Shouldn't we send a `press` event if the hold time is less than some period of time?

Reply:
Imo pushed event should be sent only when OnOff message is received. Device recognizes itself if button was hold or pushed only

Reply:
Looks like the DTH sends a pushed if it's not a hold, unless I'm reading wrong?

We want to be sending whatever the DTH would have sent.

Reply:
> Looks like the DTH sends a pushed if it's not a hold, unless I'm reading wrong?

Yes, it is

Reply:
so you'll be making a change here to send a pushed if the hold is shorter than the held threshold?

Reply:
I think right now we want to copy DTH functionality.

Reply:
Idk why, but button_utils started working properly. I've noticed that device should have also holdTime preference so I added it and rollbacked previous changes.